components/Ang/utils/sendRequest.tsx

```diff
@@ -0,0 +1,38 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import {Cache} from 'react-native-cache';
+
+const cache = new Cache({
+  namespace: 'learn-larivaar-gurbani',
+  policy: {
+    maxEntries: 50000,
+    stdTTL: 0,
+  },
+  backend: AsyncStorage,
+});
+
+export const sendRequest = (
+  page: number,
+  setCurrentAngData: Function,
+  setIsLoading: Function,
+) => {
+  setIsLoading(true);
+  cache.get(`ang-${page}`).then(data => {
+    if (data) {
+      setIsLoading(false);
+      setCurrentAngData(data);
+    } else {
+      getFromAPI(page).then(newData => {
+        setCurrentAngData(newData);
+        setIsLoading(false);
+      });
+      getFromAPI(page + 1);
+    }
+  });
+};
+
+const getFromAPI = async (page: number) => {
+  const dataRequest = await fetch(`https://api.banidb.com/v2/angs/${page}`);
```

Comment:
Recommended to put the base url of APIs in a constant variable

Reply:
Done veerji